patches/tModLoader/Terraria/DataStructures/FishingAttempt.cs.patch

```diff
@@ -0,0 +1,125 @@
++	/// The <see cref="ID.ProjectileID"/> of this bobber.
++	/// </summary>
+ 	public int bobberType;
++
++	/// <summary>
++	/// If <see langword="true"/>, this fishing attempt can catch common items.
++	/// </summary>
+ 	public bool common;
++
++	/// <summary>
++	/// If <see langword="true"/>, this fishing attempt can catch uncommon items.
++	/// </summary>
+ 	public bool uncommon;
++
++	/// <summary>
++	/// If <see langword="true"/>, this fishing attempt can catch rare items.
++	/// </summary>
+ 	public bool rare;
++
++	/// <summary>
++	/// If <see langword="true"/>, this fishing attempt can catch very rare items.
++	/// </summary>
+ 	public bool veryrare;
++
++	/// <summary>
++	/// If <see langword="true"/>, this fishing attempt can catch legendary items.
++	/// </summary>
+ 	public bool legendary;
++
++	/// <summary>
++	/// If <see langword="true"/>, this fishing attempt can catch crates.
++	/// </summary>
+ 	public bool crate;
++
++	/// <summary>
++	/// If <see langword="true"/>, this fishing attempt is in lava.
++	/// </summary>
+ 	public bool inLava;
++
++	/// <summary>
++	/// If <see langword="true"/>, this fishing attempt is in honey.
++	/// </summary>
+ 	public bool inHoney;
++
++	/// <summary>
++	/// The number of liquid tiles counted for this fishing attempt.
++	/// </summary>
+ 	public int waterTilesCount;
++
++	/// <summary>
++	/// The number of liquid tiles needed for proper fishing. If <see cref="waterTilesCount"/> is less than this, then the player will recieve a <see cref="waterQuality"/> percent debuff to their fishing power.
++	/// <br/> This debuff is automatically applied to <see cref="fishingLevel"/>.
++	/// </summary>
+ 	public int waterNeededToFish;
++
++	/// <summary>
++	/// If positive, the percent decrease in fishing power this attempt has from missing liquid tiles.
++	/// <br/> This is <strong>not</strong> how gull the body of liquid is.
++	/// </summary>
+ 	public float waterQuality;
++
++	/// <summary>
++	/// The number of chums applied to this attempt. Fishing power from chum is automatically added to <see cref="fishingLevel"/>.
++	/// </summary>
+ 	public int chumsInWater;
++
++	/// <summary>
++	/// The fishing power of this attempt after all modifications. The higher this number, the better the attempt will go.
++	/// </summary>
+ 	public int fishingLevel;
++
++	/// <summary>
++	/// If <see langword="true"/>, then this attempt can succeed if it is <see cref="inLava"/>.
++	/// </summary>
+ 	public bool CanFishInLava;
++
++	/// <summary>
++	/// How high in the sky this attempt takes place, in the range [<c>0.25f</c>, <c>1f</c>]. Any value below <c>1f</c> takes place approximately in the top 10% of the world.
++	/// <br/> The lower this value, the smaller <see cref="waterNeededToFish"/> will be, which is automatically applied.
++	/// </summary>
+ 	public float atmo;
++
++	/// <summary>
++	/// The <see cref="ID.ItemID"/> of the quest fish the Angler wants, or <c>-1</c> if this player can't catch that fish today.
```

Comment:
This is the main thing that stands out to me as an issue. You refer to item types as ItemIDs, but I think what we want to teach is the opposite. Change these sort of things to "item type" instead of pointing to the vanilla ID classes. (and the same for any other type that you refer to as TypeID)

Modders will know from the getting started guide how "X type" refers to "XID.cs" for vanilla type values and "ModContent.XType<>()" for modded type value.

Reply:
Before I change all of these, would the plain text "<x> type" or link to the associated `type` field be preferred?
Ex: "item type" vs. "<see cref="Terraria.Item.type"/>".

I saw the discussion in #collaborators a week ago about doc. standardization (https://discord.com/channels/103110554649894912/445276626352209920/1074844954381258813), but it didn't seem to reach a consensus.

Reply:
That's true, there wasn't much feedback from my question there. I think referencing Item.type is probably good, it is the primary place item types will exist. I'll need to reword some existing docs later to match our consistent terminology.

Lets go with `The item type (<see cref="Terraria.Item.type"/>) of the...` for now. I think that communicates that "item type" is the idea and Item.type is where it typically is used. The Item.type docs have more info on modded vs vanilla, so a modder should be able to click through if they forgot completely.

Reply:
Most ID references have been changed now, but I've come onto a problem with `BuffID` and `ArmorIDs`.
`ArmorIDs` has two possible vanilla fields it could point to: `Item.headSlot` or `Player.head`. I think it'd be best to use the `Item` fields (since `ModItem`s are where most modders will be using any `ArmorIDs` sets) and would like to confirm with you that that fits the established pattern.
`BuffID` doesn't seem to have any good analogue to `Item.type`. `Player.buffType` and `NPC.buffType` come to mind, but it seems clunky using them -- it'd be like referring to item types using the player's inventory.